woodwork/utils.py

```diff
@@ -298,3 +298,21 @@ def _parse_logical_type(logical_type, name):
         return logical_type
     else:
         raise TypeError(f"Invalid logical type specified for '{name}'")
+
+
+def _update_progress(start_time, current_time, progress_increment,
+                     current_progress, total, callback_function):
+    """Helper function for updating progress of a function and making a call to the progress callback
+    function, if provided. Adds the progress increment to the current progress amount and returns the
+    updated progress amount.
+
+    If provided, the callback function should accept the following parameters:
+        update: percentage change (float between 0 and 100) in progress since last call
+        progress_percent: percentage (float between 0 and 100) of total computation completed
+        time_elapsed: total time in seconds that has elapsed since start of call"""
+    new_progress = current_progress + progress_increment
+    if callback_function is not None:
+        elapsed_time = current_time - start_time
+        callback_function((progress_increment / total) * 100, (new_progress / total) * 100, elapsed_time)
+
+    return new_progress
```

Comment:
I'm not sure I understand why we still return `new_progress` even if `callback_function` is None. We don't seem to do anything with `current_progress` in the mutual info calculation other than feed it to the next call of _update_progress, so if there's no callback function, are we doing anything with this information that we're saving?

Reply:
I'm not sure now if I had a state at one point where I needed it outside the `if` block, or if I just overlooked it. In any case, it think it is safe to move the update calculation and return statement inside the `if` block.